src/migrations/20230802000000-genericReport.js

```diff
@@ -0,0 +1,2853 @@
+const {
+  REPORT_TYPE,
+  ENTITY_TYPE,
+  GOAL_STATUS,
+  OBJECTIVE_STATUS,
+  COLLABORATOR_APPROVAL_STATUSES,
+  AUDIENCE,
+  TRAINING_TYPE,
+  NEXTSTEP_NOTETYPE,
+  SOURCE_FIELD,
+} = require('../constants');
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      const sessionSig = __filename;
+      await prepMigration(queryInterface, transaction, sessionSig);
+
+      /**
+       *  create new tables for new structure:
+       * - Reports-
+       *  - ReportNationalCenters-
+       *  - ReportReasons-
+       *  - ReportTargetPopulations-
+       *  - ReportTrainingEvents-
+       *  - ReportTrainingSessions-
+       *  - ReportApprovals-
+       *  - ReportAudiences
+       *  - ReportRecipients-
+       *  - ReportCollaborators-
+       *    - ReportCollaboratorTypes-
+       *    - ReportCollaboratorRoles-
+       *  - ReportNextSteps
+       *  - ReportNextStepResources
+       *  - ReportResources-
+       *  - ReportFiles-
+       *  - ReportGoalTemplates-
+       *    - ReportGoalTemplateResources-
+       *  - ReportGoals-
+       *    - ReportGoalFieldResponse---
+       *    - ReportGoalResources-
+       *  - ReportObjectiveTemplates-
+       *    - ReportObjectiveTemplateFiles
+       *    - ReportObjectiveTemplateResources
+       *    - ReportObjectiveTemplateTopics
+       *    - ReportObjectiveTemplateTrainers
+       *  - ReportObjectives-
+       *    - ReportObjectiveFiles-
+       *    - ReportObjectiveResources-
+       *    - ReportObjectiveTopics-
+       *    - ReportObjectiveTrainers
+       *  - ReportParticipation -
+       *    - ReportParticipationParticipants
+       *  - ReportImports
+       *  - ReportPageStates
+       *
+       * additional tables needed to maintain quality data over time and maintain FOIA:
+       * - Foiaable
+       * - ValidFor-
+       * - Statuses-
+       * - Reasons-
+       * - TargetPopulations-
+       * - CollaboratorTypes-
+       * - SupportTypes -
+       * - Participants -
+       * - Audiences -
+       *
+       *  additional changes to tables that need additional columns to maintain quality
+       *  data over time and maintain FOIA:
+       * - GoalTemplates - isFoiaable
+       * - GoalTemplateResources - isFoiaable
+       * - GoalTemplateFieldPrompts - isFoiaable
+       * - Goals - isFoiaable
+       * - GoalResources - isFoiaable
+       * - GoalFieldResponses - isFoiaable
+       * - ObjectiveTemplates - isFoiaable
+       * - ObjectiveTemplateFiles - isFoiaable
+       * - ObjectiveTemplateResources - isFoiaable
+       * - ObjectiveTemplateTopics - isFoiaable
+       * - Objectives - isFoiaable
+       * - ObjectiveFiles - isFoiaable
+       * - ObjectiveResources - isFoiaable
+       * - ObjectiveTopics - isFoiaable
+       *  */
+
+      //---------------------------------------------------------------------------------
+      await queryInterface.createTable('Foiaable', {
```

Comment:
Prob havent gotten there yet but how do we use Foiaable and validFor tables?

Reply:
Foiaable table identifies which columns for which tables are "read-only" after the isFoiaable flag is set. This allows some values to still be updated while maintaining the integrity of other columns.

ValidFor is used in conjunction with many of the enum-esk tables to allow similar enums to be located in the same table. Like "Statuses" hold the valid statuses for Training Events, Training Sessions, Goals, Objectives, and Collaborators. So they all reside in the same table and have the mapsTo construct to allow extensibility without each requiring their own table. 

Future-future plan would be to have an admin page to  manage these enum-esk values. Think Audience, national centers, participants, reasons, roles, statuses, support types, target populations, and topics. Let them manage the values. Then the UI and data just run off the admin maintained data. Again, future-future plan.